src/testing/advection_explicit_periodic.cpp

```diff
@@ -10,180 +10,903 @@
+
+
+            pcout << " Grid size h: " << dx 
+                 << " L2-soln_error: " << l2error_mpi_sum
+                 << " Linf-soln_error: " << linferror_mpi
+                 << " Residual: " << ode_solver->residual_norm
+                 << std::endl;
+
+
+            if (igrid > igrid_start) {
+                const double slope_soln_err = log(soln_error[igrid]/soln_error[igrid-1])
+                                      / log(grid_size[igrid]/grid_size[igrid-1]);
+                const double slope_soln_err_inf = log(soln_error_inf[igrid]/soln_error_inf[igrid-1])
+                                      / log(grid_size[igrid]/grid_size[igrid-1]);
+              //  const double slope_output_err = log(output_error[igrid]/output_error[igrid-1])
+     //                                 / log(grid_size[igrid]/grid_size[igrid-1]);
+                pcout << "From grid " << igrid-1
+                     << "  to grid " << igrid
+                     << "  dimension: " << dim
+                     << "  polynomial degree p: " << poly_degree
+                     << std::endl
+                     << "  solution_error1 " << soln_error[igrid-1]
+                     << "  solution_error2 " << soln_error[igrid]
+                     << "  slope " << slope_soln_err
+                     << "  solution_error1_inf " << soln_error_inf[igrid-1]
+                     << "  solution_error2_inf " << soln_error_inf[igrid]
+                     << "  slope " << slope_soln_err_inf
+                     << std::endl;
+            
+                if(igrid == n_grids-1){
+                    if(std::abs(slope_soln_err_inf-(poly_degree+1))>0.1)
+                        return 1;
+                }            
+
+            }
+
+    }
+        pcout << " ********************************************"
+             << std::endl
+             << " Convergence rates for p = " << poly_degree
+             << std::endl
+             << " ********************************************"
+             << std::endl;
+        convergence_table.evaluate_convergence_rates("soln_L2_error", "cells", dealii::ConvergenceTable::reduction_rate_log2, dim);
+        convergence_table.evaluate_convergence_rates("soln_Linf_error", "cells", dealii::ConvergenceTable::reduction_rate_log2, dim);
+        convergence_table.evaluate_convergence_rates("output_error", "cells", dealii::ConvergenceTable::reduction_rate_log2, dim);
+        convergence_table.set_scientific("dx", true);
+        convergence_table.set_scientific("soln_L2_error", true);
+        convergence_table.set_scientific("soln_Linf_error", true);
+        convergence_table.set_scientific("output_error", true);
+        if (pcout.is_active()) convergence_table.write_text(pcout.get_stream());
+        }//end of OOA
+
+       // convergence_table_vector.push_back(convergence_table);
+
+	return 0; //need to change
 }
 
 //int main (int argc, char * argv[])
```

Comment:
Delete all the comments below. No need to keep dead code.

Reply:
fixed